### DISCUSSION TITLE
Target of URI doesn't exist: 'package:flutter_gen/gen_l10n/gallery_localizations.dart'

### DISCUSSION THREAD

**Discussion by arrullin** [community/community#57903] [Programming Help] [answered]
### Select Topic Area

Question

### Body

I am now using flutter gallary in my project, this is the package reference:

```js
import 'package:flutter_gen/gen_l10n/gallery_localizations.dart';
```

but it shows:

```bash 
Target of URI doesn't exist: 'package:flutter_gen/gen_l10n/gallery_localizations.dart'.
```

**Comment by santosharron** [comment, marked as answer]
Install the `f;utter_gen` properly. Check out this [documentation](https://pub.dev/packages/flutter_gen) for more info.

**Installation** 
Works with macOS, Linux and Windows.

```bash
dart pub global activate flutter_gen
```

Then, Run FlutterGen
```bash
flutter packages pub run build_runner build
```

Once Everything is installed You can find the **Build** Folder in `./.dart_tools`.

![image](https://github.com/community/community/assets/73644573/386d1479-a609-4388-a37c-d05ab9a0c939)

You might need to structure it accordingly by placing the `flutter_gen` in your `lib` folder and change the import statement from this.

```bash
import 'package:flutter_gen/gen_l10n/gallery_localizations.dart'.
```
To this

```bash
import 'package:{PROJECT_NAME}/flutter_gen/gen_l10n/gallery_localizations.dart'.
```

If That's not working Try closing the IDE(Android studio or VS Code) and open it again, and the issue will be resolved!

**Reply by arrullin** [reply]
It Worked!! Thank you so much.

**Comment by fulstadev** [comment]
Came across this, had the same problem and noticed that this is actually no longer the correct solution to the problem. Depends on your flutter version, but since a stable release dropped post 3.32.0 (which is more than a year ago by the time of writing this comment), `package:flutter_gen` support has been [dropped](https://docs.flutter.dev/release/breaking-changes/flutter-generate-i10n-source). 

The correct way to solve this issue is thus to no longer rely on importing your app localizations through a `package:flutter_gen` import, but instead:

1. As the [migration docs](https://docs.flutter.dev/release/breaking-changes/flutter-generate-i10n-source#migration-guide) mention, [configure](https://docs.flutter.dev/ui/internationalization#configuring-the-l10n-yaml-file) your `l10n.yml` file to use `synthetic-package: false` ([default](https://docs.flutter.dev/ui/internationalization#configuring-the-l10n-yaml-file) is `true`). This will tell flutter that the localizations class used in your app is actually part of your flutter code, and that it cannot be imported through some `flutter_gen` path.

2. Then, I'd recommend you to explicitly specify an `output-dir` [config](https://docs.flutter.dev/ui/internationalization#configuring-the-l10n-yaml-file) value, ideally for a folder within your `/lib`, e.g. `lib/l10n`. This will tell the `flutter-gen-l10n` program where to generate the `AppLocalizations` classes of your `.arb` files. And note that the specification of this l10n config actually requires the `synthetic-package: false` flag to be set, according to the [docs](https://docs.flutter.dev/ui/internationalization#configuring-the-l10n-yaml-file) (which you already did if you completed step 1 above).

3. From now on, you can go on ahead and import your `AppLocalizations` into whichever file by using local project paths. Given the provided example above, the correct import statement would now be `package:{name-value-in-your-pubspec}/l10n/app_localizations.dart`. Feel free to use relative import paths too, but that's [subjective](https://stackoverflow.com/questions/59693195/flutter-imports-relative-path-or-package). I highly prefer absolute paths as my IDE's refactorings automatically understand these, but not the relative ones.

Note: I'd recommend you to define your `output-dir` within `lib` even if you work on earlier flutter versions, as that allows your IDE to most likely faster detect the `AppLocalizations` class for auto-importing, as the class will be where all your other flutter apps classes sit (that is, within `lib`, as long as you used the paths provided in the example above). It also avoids the app localizations issue whenever you upgrade your flutter version, which is a very well known and superannoying issue, unfortunately.
